source/collimation.f90

```diff
@@ -986,108 +785,81 @@ subroutine collimate_init()
+  if(rnd_seed == 0) rnd_seed = mclock_liar()
+  if(rnd_seed <  0) rnd_seed = abs(rnd_seed)
   rnd_lux = 3
   rnd_k1  = 0
   rnd_k2  = 0
   call rluxgo(rnd_lux, rnd_seed, rnd_k1, rnd_k2)
-!  call recuin(rnd_seed, 0)
   write(outlun,*) 'INFO>  rnd_seed: ', rnd_seed
 
-!Call distribution routines only if collimation block is in fort.3, otherwise
-!the standard sixtrack would be prevented by the 'stop' command
+  ! Call distribution routines only if collimation block is in fort.3
+  cdist_logUnit = outlun
+  cdist_energy  = myenom
+  cdist_alphaX  = myalphax
+  cdist_alphaY  = myalphay
+  cdist_betaX   = mybetax
+  cdist_betaY   = mybetay
+  cdist_emitX   = myemitx0_dist
+  cdist_emitY   = myemity0_dist
   if(radial) then
-    call makedis_radial(myalphax, myalphay, mybetax, mybetay, myemitx0_dist, myemity0_dist, &
-                        myenom, nr, ndr, myx, myxp, myy, myyp, myp, mys)
+    call cdist_makeRadial
   else
-    select case(do_thisdis)
-    case(0)
-      continue
-    case(1)
-      call makedis(myalphax, myalphay, mybetax, mybetay, myemitx0_dist, myemity0_dist, &
-                   myenom, mynex, mdex, myney, mdey, myx, myxp, myy, myyp, myp, mys)
-    case(2)
-      call makedis_st(myalphax, myalphay, mybetax, mybetay, myemitx0_dist, myemity0_dist, &
-                      myenom, mynex, mdex, myney, mdey, myx, myxp, myy, myyp, myp, mys)
-    case(3)
-      call makedis_de(myalphax, myalphay, mybetax, mybetay, myemitx0_dist, myemity0_dist, &
-                      myenom, mynex, mdex, myney, mdey,myx, myxp, myy, myyp, myp, mys,enerror,bunchlength)
-    case(4)
-      call readdis(filename_dis, myx, myxp, myy, myyp, myp, mys)
-    case(5)
-      call makedis_ga(myalphax, myalphay, mybetax, mybetay, myemitx0_dist, myemity0_dist, &
-                      myenom, mynex, mdex, myney, mdey, myx, myxp, myy, myyp, myp, mys, enerror, bunchlength )
-    case(6)
-      call readdis_norm(filename_dis, myalphax, myalphay, mybetax, mybetay, &
-                        myemitx0_dist, myemity0_dist, myenom, myx, myxp, myy, myyp, myp, mys, enerror, bunchlength)
-    case default
-      write(lerr,"(a)") "COLL> ERROR Review your distribution parameters!"
-      call prror
-    end select
+    call cdist_makeDist(do_thisdis)
   end if
 
-!++  Reset distribution for pencil beam
-!
-  if(ipencil.gt.0) then
+  ! Reset distribution for pencil beam
+  if(ipencil > 0) then
     write(lout,"(a)") "COLL> WARNING Distributions reset to pencil beam!"
-    write(outlun,*) 'WARN>  Distributions reset to pencil beam!'
-    do j = 1, napx
-      myx(j)  = zero
-      myxp(j) = zero
-      myy(j)  = zero
-      myyp(j) = zero
-    end do
+    write(outlun,*)   "WARN> Distributions reset to pencil beam!"
+    xv1(1:napx) = zero
+    yv1(1:napx) = zero
+    xv2(1:napx) = zero
+    yv2(1:napx) = zero
   endif
 
   ! Optionally write the generated particle distribution
-  if(dowrite_dist .and. do_thisdis /= 0) then
+  if(dowrite_dist) then
```

Comment:
why removing the check on `do_thisdis`?

Reply:
I see no reason why you shouldn't be able to dump the distribution even if you're not using the internal generator. You may still want to know what it looks like.

Reply:
yes, but then one may want to do this no matter if the collimation block has been issued or not

Reply:
Not sure I follow.

I added this check when I added format 0 as a valid option, but on second thought I don't really see any reason to block the user from generating the dist files when they have requested them (for whatever reason). This also blocks the user from dumping a radial dist.